### PR TITLE
(GH-1231) Use DirectoryInfo to evaluate CacheLocation

### PR DESCRIPTION
**Pull request by mwrock**
The former fix (GH-1210) fails when the CacheLocation ends with a slash. This uses a DirectoryInfo instance to examine the last directory in the path in a more robust manner

### CODE REVIEW

**Comment by mwrock** [comment]
I just tested the appveyor drop with boxstarter and this looks to be working.

**Comment by mwrock** [comment]
fixes #1231

File: src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs [hunk 1]
```diff
@@ -199,8 +200,8 @@ private static void set_config_items(ChocolateyConfiguration config, ConfigFileS
                 config.CacheLocation = fileSystem.get_temp_path(); // System.Environment.GetEnvironmentVariable("TEMP");
                 // TEMP gets set in EnvironmentSettings, so it may already have 
                 // chocolatey in the path when it installs the next package from
-                // the API. 
-                if(!config.CacheLocation.EndsWith("chocolatey")) {
+                // the API.
+                if(new DirectoryInfo(config.CacheLocation).Name.ToLower() != "chocolatey") {
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs]: Use `fileSystem.get_directory_info()`  and use either `.to_lower()` or `Name.is_equal_to()`. The latter is preferred.

**Comment by mwrock** [comment]
hmm this is dying with `'string' does not contain a definition for 'is_equal_to'` but that seems to be a valid extension method used throughout this file. My c# is very rusty. Will look more later tonight.

**Comment by ferventcoder** [comment]
> hmm this is dying with 'string' does not contain a definition for 'is_equal_to' but that seems to be a valid extension method used throughout this file. My c# is very rusty. Will look more later tonight.

My guess?

`new DirectoryInfo(config.CacheLocation).Name.ToLower() != "chocolatey"` was changed to `fileSystem.get_directory_info(config.CacheLocation).Name.is_equal_to() != "chocolatey"` instead of `!fileSystem.get_directory_info(config.CacheLocation).Name.is_equal_to("chocolatey")`. The first change would definitely produce the error, it's needs a value passed to the method.

**Comment by mwrock** [comment]
well `!fileSystem.get_directory_info(config.CacheLocation).Name.is_equal_to("chocolatey")` is exacty what's in this pr.

**Comment by mwrock** [comment]
actually its `get_directory_info_for` not `get_directory_info` but I think thats right.

**Comment by ferventcoder** [comment]
So it is. So it is.

**Comment by mwrock** [comment]
ok. I just ran `MSBuild.exe C:\dev\choco\src\chocolatey\chocolatey.csproj /t:Build` and it compiles just fine. So I'm not entirely sure why that fails in appveyor. The appveyor log is huge and theer might be something in there I am missing.

BTW: the build.bat breaks on nuget.exe in the choco repo. If I swap in the latest nuget.exe the nuget installs work but then the build fails in ilmerge.exe. I think the nuget problems are due to API incompatibilities with older nuget clients (just an educated guess).

Anyways I found that vanilla msbuild ended up being the easiest way for me to locally sanity check the build.

**Comment by mwrock** [comment]
Looking again at the appveyor log I actually think this is a test setup issue and not related directly to the config builder itself. Perhaps somthing needs to be imported or added to a container.

**Comment by ferventcoder** [comment]
Don't get too hung up on the appveyor issues. Also, as a committer, you may think about using stable and not master.

**Comment by mwrock** [comment]
closing in favor of #1233 targeting stable

**Pull request closed without merging**